source/material_model/rheology/visco_plastic.cc

```diff
@@ -471,9 +486,9 @@ namespace aspect
                            "Stabilizes strain dependent viscosity. Units: \\si{\\per\\second}.");
         prm.declare_entry ("Reference strain rate","1.0e-15",Patterns::Double (0.),
                            "Reference strain rate for first time step. Units: \\si{\\per\\second}.");
-        prm.declare_entry ("Minimum viscosity", "1e17", Patterns::Double (0.),
+        prm.declare_entry ("Minimum viscosity", "1e17", Patterns::Anything(),
```

Comment:
Here and a couple of lines down, you could write this as
```suggestion
        prm.declare_entry ("Minimum viscosity", "1e17", Patterns::List(Patterns::Double(0.), 1),
```
meaning it's at least one entry in the list, and each entry is a positive number.

Reply:
I think this type may not match  this type of entry:
`set Minimum viscosity = background: 1.e17 | 1.e17 | 1.e19, right: 1.e16`
The error message I got is:
    Line <90> of file <input string>:
    The entry value 
        background: 1.e17 | 1.e17 | 1.e19, right: 1.e16
    for the entry named
        Minimum viscosity
    does not match the given pattern:
        [List of <[Double 0...MAX_DOUBLE (inclusive)]> of length 1...4294967295 (inclusive)]